File: src/main/java/de/viktorlevin/starkeverbenbot/service/alltypes/ExampleService.java
package de.viktorlevin.starkeverbenbot.service.alltypes;

import de.viktorlevin.starkeverbenbot.client.ExamplesClient;
import de.viktorlevin.starkeverbenbot.dto.ExamplesDto;
import de.viktorlevin.starkeverbenbot.service.StarkeVerbenService;
import de.viktorlevin.starkeverbenbot.service.TextService;
import de.viktorlevin.starkeverbenbot.service.WortService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import org.telegram.telegrambots.meta.api.methods.BotApiMethod;
import org.telegram.telegrambots.meta.api.objects.CallbackQuery;

import java.util.List;

@Slf4j
@Service
@RequiredArgsConstructor
public class ExampleService {
    private final ExamplesClient examplesClient;
    private final WortService wortService;
    private final StarkeVerbenService starkeVerbenService;
    private final TextService textService;

    public List<BotApiMethod> getExample(CallbackQuery callbackQuery) {
        Long chatId = callbackQuery.getMessage().getChatId();
        String username = callbackQuery.getFrom().getUserName();
        String data = callbackQuery.getData();
        log.info("Got CallbackExample message {} from {} with username {}",
                data, chatId, username);
        try {
            ExamplesDto example = examplesClient.getExample(extractWord(data));
            return List.of(textService.messageWithExample(example, chatId));
        } catch (Exception exception) {
            log.error("error during getting examples", exception);
            throw new RuntimeException("Не могу найти примеры для этого слова...");
        }
    }

    private String extractWord(String data) {
        Integer id = Integer.valueOf(data.split(":")[1]);
        if (data.contains("Verb")) {
            return starkeVerbenService.findStarkesVerbById(id).getPartizip2();
        } else {
            return wortService.findWortById(id).getWord();
        }
    }
}

File: downloader/src/main/java/de/viktorlevin/voicedownloader/service/ExampleService.kt
package de.viktorlevin.voicedownloader.service

import de.viktorlevin.voicedownloader.client.ExamplesClient
import de.viktorlevin.voicedownloader.dto.ExamplesDto
import lombok.extern.slf4j.Slf4j
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service
import java.util.*

@Slf4j
@Service
class ExampleService(private val examplesClient: ExamplesClient) {
    private val random: Random = Random()

    companion object {
        const val LANGUAGE = "de-ru"
    }

    private val log: Logger = LoggerFactory.getLogger(ExampleService::class.java)

    fun getExamples(word: String): ExamplesDto.Example {
        log.info("Getting examples for {}", word)
        val (result) = examplesClient.getExamples(word, LANGUAGE)
        log.info("Examples for {} successfully received", word)
        val examplesList = result[0].examples
        return examplesList[random.nextInt(examplesList.size)]
    }
}